public/inbox.php

```diff
@@ -178,14 +176,18 @@ function MarkAsUnread(msgID) {
             echo "<div class='float-right'>";
 
             if ($offset > 0) {
-                echo "<a class='btn btn-link' href='/inbox.php?o=" . ($offset - $maxCount) . "&amp;u=$unreadOnly&amp;s=$outbox'>";
-                echo "&lt; Previous $maxCount";
+                $newOffset = max($offset - $count, 0);
+                echo "<a class='btn btn-link' href='/inbox.php?s=$outbox&u=$unreadOnly&c=$count&o=$newOffset'>";
+                echo "&lt; Previous $count";
                 echo "</a>";
             }
 
-            if ($totalMsgs == $maxCount) {
-                echo "<a class='btn btn-link' href='/inbox.php?o=" . ($offset + $maxCount) . "&amp;u=$unreadOnly&amp;s=$outbox'>";
-                echo "Next $maxCount &gt;";
+            $messagesLeft = ($unreadOnly ? $unreadMessageCount : $totalMessageCount) - $offset;
+            if ($messagesLeft > $count) {
+                $newOffset = $offset + $count;
+                $messagesNext = min($count, $messagesLeft - $count);
+                echo "<a class='btn btn-link' href='/inbox.php?s=$outbox&u=$unreadOnly&c=$count&o=$newOffset'>";
+                echo "Next $messagesNext &gt;";
                 echo "</a>";
```

Comment:
Consider swapping out the prev/next links for a call to `RenderPaginator`. It would improve visual consistency with other parts of the product and assist with some of the edge cases (like attempted negative navigation).

Reply:
Nice. It didn't cross my mind `inbox.php` was still using the old nav solution.